### PR TITLE
Allow multiple dds files for CubeMap

### LINKED ISSUE #3188: Cube Map DDS issues
Single file DDS doesn't parse correctly; for compressed texture `bbs` is always `0` so `dataOffset` doesn't advance:

https://github.com/BabylonJS/Babylon.js/blob/e3b20e41f69b3aa7f43269ae08126eb9db5d00b4/src/Tools/babylon.dds.ts#L520

Multi-file DDS ignores the `files` parameter  only loading single `rootUrl`
https://github.com/BabylonJS/Babylon.js/blob/e3b20e41f69b3aa7f43269ae08126eb9db5d00b4/src/Engine/babylon.engine.ts#L3933

### PR DESCRIPTION
**Pull request by benaadams**
Resolves #3188

### CODE REVIEW

**Comment by benaadams** [comment]
Still has some weird artifacts I'm trying to track down

skipping

**Comment by deltakosh** [comment]
OK waiting for your go here

**Comment by benaadams** [comment]
Think is same issue; is because the `dataOffset` isn't advancing for each `mip` level; due to `bpp` being 0

**Comment by benaadams** [comment]
K, this fixes both issues can use just root, root + extensions or files

**Comment by deltakosh** [comment]
huge thanks!

File: src/Tools/babylon.dds.ts [hunk 1]
```diff
@@ -517,13 +517,18 @@
                             engine._uploadCompressedDataToTexture(sampler, i, internalFormat, width, height, byteArray);
                         }
                     }
-                    dataOffset += width * height * (bpp / 8);
+                    dataOffset += dataLength;
```
Comment (deltakosh) [inline src/Tools/babylon.dds.ts]: Actually this does not work for non compressed DDS
I'll push a fix to update to:
dataOffset += bpp ? (width * height * (bpp / 8)) : dataLength;

Comment (benaadams) [reply src/Tools/babylon.dds.ts]: I never knew there was such a thing as non-compressed DDS :)

Comment (deltakosh) [reply src/Tools/babylon.dds.ts]: Pretty useful as compressed DDS is not wide supported

**Pull request merged**